Feature selector api

mollybsmith-noaa opened the pull request:
This pull request contains my API for getting scorecard metadata from individual apps. It uses server-side routes which are in the linked matsMethods.js commits. Documentation fo the API is at https://docs.google.com/document/d/1nX_hB0P9VyUokSYxWwtHwmB6MrTtklX0ZFlgOUtkyEg/edit?usp=sharing

randytpierce reviewed (approved):
I think this all looks good.

ian-noaa commented on apps/precipGauge/server/main.js:
```diff
@@ -384,22 +382,6 @@ const doCurveParams = function () {
         );
     }
 
-    if (matsCollections["variable"].findOne({name: 'variable'}) == undefined) {
-        matsCollections["variable"].insert(
-            {
-                name: 'variable',
-                type: matsTypes.InputTypes.select,
-                options: variables,
-                controlButtonCovered: true,
-                default: variables[0],
-                unique: false,
-                controlButtonVisibility: 'none',
-                displayOrder: 2,
-                displayPriority: 1,
-                displayGroup: 1
-            });
-    }
-
```
Just curiosity - were these moved elsewhere? I wasn't finding a place they were refactored to so I'm wondering if they were vestigial code.

mollybsmith-noaa replied:
Well, my reply did not go to the right thread. Sigh. Here it is again in the correct place for posterity:

"No, they shouldn’t have been there in the first place. I added them
previously thinking they’d make an eventual scorecard easier, when in fact
they did the opposite, so I just removed them (those are single-variable
apps). Thanks for reviewing the pull request!"

mollybsmith-noaa replied:
Basically, it was code added specifically for the scorecard which the scorecard did not end up needing.

ian-noaa replied:
Makes sense. Thanks!

ian-noaa reviewed (approved):
Looks great! Thanks for the commit structure in this PR, Molly. It certainly made it easier to review!

mollybsmith-noaa commented:
No, they shouldn’t have been there in the first place. I added them
previously thinking they’d make an eventual scorecard easier, when in fact
they did the opposite, so I just removed them (those are single-variable
apps). Thanks for reviewing the pull request!

On Wed, Sep 7, 2022 at 4:24 PM Ian McGinnis ***@***.***>
wrote:

> ***@***.**** approved this pull request.
>
> Looks great! Thanks for the commit structure in this PR, Molly. It
> certainly made it easier to review!
> ------------------------------
>
> In apps/precipGauge/server/main.js
> <https://github.com/NOAA-GSL/MATS/pull/766#discussion_r965347998>:
>
> > -    if (matsCollections["variable"].findOne({name: 'variable'}) == undefined) {
> -        matsCollections["variable"].insert(
> -            {
> -                name: 'variable',
> -                type: matsTypes.InputTypes.select,
> -                options: variables,
> -                controlButtonCovered: true,
> -                default: variables[0],
> -                unique: false,
> -                controlButtonVisibility: 'none',
> -                displayOrder: 2,
> -                displayPriority: 1,
> -                displayGroup: 1
> -            });
> -    }
> -
>
> Just curiosity - were these moved elsewhere? I wasn't finding a place they
> were refactored to so I'm wondering if they were vestigial code.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-GSL/MATS/pull/766#pullrequestreview-1099887231>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHBWGZL56CYKR4K6NNFP5BLV5EI2TANCNFSM6AAAAAAQHADYCY>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>
